controllers/challengeController.js

```diff
@@ -3,37 +3,49 @@ const challengeQuery = require('../models/challenges')
 const ERROR_MESSAGE = 'Something went wrong. Please try again or contact admin'
 
 /**
- * Get the challenges or add the challenge
+ * Get the challenges
  * @param {Object} req - Express request object
  * @param {Object} res - Express response object
  */
 
-const sendChallengeResponse = async (req, res) => {
+const fetchChallenges = async (req, res) => {
+  const allChallenges = await challengeQuery.fetchChallenges()
+
+  if (allChallenges.length > 0) {
+    return res.json({
+      message: 'Challenges returned successfully!',
+      challenges: allChallenges
+    })
+  } else {
+    return res.boom.notFound('No challenges found')
+  }
+}
+
+/**
+ * Add a challenge
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+
+const createChallenge = async (req, res) => {
   try {
-    if (req.method === 'GET') {
-      const allChallenges = await challengeQuery.fetchChallenges()
-      if (allChallenges.length > 0) {
-        return res.status(200).json({
-          message: 'Challenges returned successfully!',
-          challenges: allChallenges
-        })
-      } else {
-        return res.boom.notFound('No challenges found')
-      }
-    } else {
-      if (req.method === 'POST') {
-        const challengeAdded = await challengeQuery.postChallenge(req.body)
-        if (challengeAdded) {
-          return res.status(200).json({
-            message: 'Challenge added successfully',
-            challenges: challengeAdded
-          })
-        }
-      } else {
-        return res.boom.notFound('Unable to add challenge')
-      }
+    const { title, level, start_date: startDate, end_date: endDate } = req.body
+    if (!title || !level || !startDate || !endDate) {
+      return res.boom.badData('Empty fields received!')
+    }
```

Comment:
Please make a middleware for validating. The flow wouldn't reach the controller if there are wrong/missing fields.